src/components/App.js

```diff
@@ -120,54 +125,37 @@ const App = () => {
   };
 
   const assignStartLocation = (latlng) => {
-    clearPath();
-    setStartLocation(latlng);
-    setStartAddress('');
+    dispatch(clearPath());
+    dispatch(setStartLocation(latlng));
+    dispatch(setStartAddress(''));
 
     reverseGeocode(latlng).then((address) => {
-      if (!address) {
+      if (!address)
         return handleError(new Error('Unable to get reverse geocoding result.'));
-      }
 
-      setStartAddress(address);
+      dispatch(setStartAddress(address));
     });
   };
 
   const assignEndLocation = (latlng) => {
-    clearPath();
-    setEndLocation(latlng);
-    setEndAddress('');
+    dispatch(clearPath());
+    dispatch(setEndLocation(latlng));
+    dispatch(setEndAddress(''));
 
     reverseGeocode(latlng).then((address) => {
       if (!address) {
         return handleError('Unable to get reverse geocoding result.');
       }
 
-      setEndAddress(address);
+      dispatch(setEndAddress(address));
     });
   };
 
   const updateControls = (items) => {
-    if (items.startLocation) {
-      assignStartLocation(items.startLocation);
-    }
-
-    if (items.scenario) {
-      setScenario(items.scenario);
-    }
-
-    if (items.startAddress !== undefined) {
-      setStartAddress(items.startAddress);
-    }
-
-    if (items.endAddress !== undefined) {
-      setEndAddress(items.endAddress);
-    }
-  };
-
-  const clearRoute = () => {
-    clearPath();
-    clearMarkers();
+    if (items.startLocation) assignStartLocation(items.startLocation);
+    if (items.scenario) setScenario(items.scenario);
+    if (items.startAddress !== undefined) dispatch(setStartAddress(items.startAddress));
```

Comment:
I think I saw somewhere (search.js?) `startAddress` initialized to `null` rather than `undefined`. Worth changing this (and other similar patterns?) to a more generic falsey check?
```suggestion
    if (!items.startAddress) dispatch(setStartAddress(items.startAddress));
```